Dockerfile

```diff
@@ -15,16 +15,27 @@ WORKDIR /app
 
 RUN /root/.local/bin/pipenv sync
 
+COPY .git .git
+RUN echo $(git rev-parse --short HEAD) > .version
+RUN echo $(date -Is) >> .version
+
 FROM python:3.10-slim AS runtime
 
 WORKDIR /app
 
 # copy venv into runtime
 COPY --from=builder /app/.venv/ /app/.venv/
+COPY --from=builder /app/.version /app/.version
+
+# keep container size down + we don't need it
+RUN rm -rf .git
```

Comment:
May be better to copy everything but `.git` instead of copying and removing it. Not sure what utilities this image has by default tho (`ls` into `grep` into `cp` at worst, but `find -exec` or `rsync` are better)

Reply:
you're absolutely right in terms of how layers work but it's not easy to do. Might change it so it only copies source directories.